include/fiction/io/write_sqd_sim_result.hpp

```diff
@@ -0,0 +1,270 @@
+}
+
+template <typename Lyt>
+class write_sqd_sim_result_impl
+{
+  public:
+    write_sqd_sim_result_impl(const sidb_simulation_result<Lyt>& result, std::ostream& s) : sim_result{result}, os{s} {}
+
+    void run()
+    {
+        os << siqad::XML_HEADER << siqad::OPEN_SIM_OUT;
+
+        write_engine_info();
+
+        write_simulation_parameters();
+
+        write_physical_locations();
+
+        write_electron_distributions();
+
+        os << siqad::CLOSE_SIM_OUT;
+    }
+
+  private:
+    /**
+     * The simulation result to write.
+     */
+    const sidb_simulation_result<Lyt>& sim_result;
+    /**
+     * The output stream to write to.
+     */
+    std::ostream& os;
+    /**
+     * Writes the engine information to the output stream in XML format.
+     */
+    void write_engine_info()
+    {
+        os << fmt::format(siqad::ENG_INFO_BLOCK, sim_result.algorithm_name, FICTION_VERSION, FICTION_REPO, 0,
+                          fmt::format("{:%Y-%m-%d %H:%M:%S}", fmt::localtime(std::time(nullptr))),
+                          sim_result.simulation_runtime.count());
+    }
+
+    /**
+     * Writes both the `physical_parameters` as well as the `additional_simulation_parameters` to the output stream in
+     * XML format.
+     */
+    void write_simulation_parameters()
+    {
+        os << siqad::OPEN_SIM_PARAMS;
+
+        // physical parameters
+        os << fmt::format(siqad::PHYS_SIM_PARAMS, sim_result.physical_parameters.lambda_tf,
+                          sim_result.physical_parameters.epsilon_r, sim_result.physical_parameters.mu);
+
+        // additional simulation parameters
+        std::for_each(sim_result.additional_simulation_parameters.cbegin(),
+                      sim_result.additional_simulation_parameters.cend(),
+                      [this](const auto& p)
+                      {
+                          const auto& [name, value] = p;
+
+                          if (value.has_value())
+                          {
+                              os << fmt::format(siqad::ADD_SIM_PARAM, name, any_to_string(value), name);
+                          }
+                      });
+
+        os << siqad::CLOSE_SIM_PARAMS;
+    }
+    /**
+     * Writes the physical locations of all SiDBs to the output stream in XML format.
+     */
+    void write_physical_locations()
+    {
+        os << siqad::OPEN_PHYSLOC;
+
+        // only write locations if any charge distributions are present
+        if (!sim_result.charge_distributions.empty())
+        {
+            // take the first distribution as the reference layout
+            const auto& lyt = sim_result.charge_distributions.front();
+
+            lyt.foreach_cell(  // this should ensure the same order for all subsequent distributions
+                [this](const auto& c)
+                {
+                    const auto [nm_x, nm_y] = sidb_nm_position<Lyt>(sim_result.physical_parameters, c);
```

Comment:
Since ``SiQAD`` expects the locations to be in ``Å``, we need to add a factor of ``0.1``. But before we have to merge the new PR.

Reply:
Got it, thank you!